examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,114 @@
+use bevy::{prelude::*, tasks::{AsyncComputeTaskPool, Task}};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use futures_lite::future;
+
+/// This example shows how to use the ECS and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: u32 = 6;
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spans multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(
+    mut commands: Commands,
+    thread_pool: Res<AsyncComputeTaskPool>,
+) {
+    for x in 0 ..NUM_CUBES {
+        for y in 0 ..NUM_CUBES {
+            for z in 0 ..NUM_CUBES {
+
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {
+                        // Spinning for 'duration', simulating doing hard

```

Comment:
Instead of spinning in a busy loop, perhaps we *should* do some compute and print it at the end so it's a bit more realistic? I'd probably do something like calculate the Nth fibonacci number or something as "generic example compute".

Reply:
Hmm, I'm not sure I completely agree as adding some Fibonacci calculations would, I feel, just add extra noise to the code and complicate it. 

If we really wanted to have real work be done, have it be 'real' game calculations would be a little better.

Though, I don't particularly feel super strong either way.

Reply:
The problem is "real game calculations" are usually rather long, and not super easy to understand. On the other hand, fibonacci numbers can both be explained, and implemented in a very small number of lines. I'm not 100% sold on them either, but I couldn't think of a shorter "compute" example than that.

Reply:
Here I was following the examples in the ``bevy_tasks`` crate:

https://github.com/bevyengine/bevy/blob/97d8e4e1793ede3df8c77ed44736e800b38ff7a4/crates/bevy_tasks/examples/busy_behavior.rs#L18-L20